src/overlays/actors/ovl_En_Mm/z_en_mm.c

```diff
@@ -34,38 +32,190 @@ const ActorInit En_Mm_InitVars = {
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    if (this->actor.parent != NULL) {
+        func_80965BBC(this);
+        return;
+    }
+    if (this->actor.cutscene >= 0) {
+        action = func_80665D3C;
+    } else {
+        action = func_80965DB4;
+    }
+    EnMm_SetupAction(this, action);
+}
+
+void EnMm_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnMm* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80665D3C(EnMm* this, GlobalContext* globalCtx) {
+    s16 cutscene = ActorCutscene_GetAdditionalCutscene(this->actor.cutscene);
+
+    if (ActorCutscene_GetCanPlayNext(cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(cutscene, &this->actor);
+        EnMm_SetupAction(this, func_80965DB4);
+    } else {
+        ActorCutscene_SetIntentToPlay(cutscene);
+    }
+}
+
+void func_80965DB4(EnMm* this, GlobalContext* globalCtx) {
+    s16 direction;
+    Vec3f sp50;
+    s16 unused;
+    f32 temp_f14;
+    f32 temp_f12;
+    f32 temp_f2;
+    s32 angle;
+    s32 pad;
+
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        func_80965BBC(this);
+        Audio_PlayActorSound2(&this->actor, NA_SE_PL_PULL_UP_ROCK);
+    } else {
+        if ((this->actor.velocity.y > 0.0f) && (this->actor.bgCheckFlags & 0x10)) {
+            this->actor.velocity.y = 0.0f;
+        }
+
+        if ((this->actor.speedXZ != 0.0f) && (this->actor.bgCheckFlags & 8)) {
+            angle = BINANG_SUB(this->actor.world.rot.y, BINANG_ROT180(this->actor.wallYaw));
+            this->actor.world.rot.y += BINANG_SUB(0x8000, (s16)(angle * 2));
+            this->actor.speedXZ *= 0.5f;
+            CollisionCheck_SpawnShieldParticles(globalCtx, &this->actor.world.pos);
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_HUMAN_BOUND);
+        }
+
+        if (!(this->actor.bgCheckFlags & 1)) {
+            Math_StepToF(&this->actor.speedXZ, 0.0f, 0.08f);
+        } else {
+            temp_f14 = Math_SinS(this->actor.world.rot.y) * this->actor.speedXZ;
+            temp_f12 = Math_CosS(this->actor.world.rot.y) * this->actor.speedXZ;
+            func_800B75A0(this->actor.floorPoly, &sp50, &unused);
+            temp_f14 += 3.0f * sp50.x;
+            temp_f12 += 3.0f * sp50.z;
+            temp_f2 = sqrtf(SQ(temp_f14) + SQ(temp_f12));
+
+            if ((temp_f2 < this->actor.speedXZ) ||
+                (func_800C9C74(&globalCtx->colCtx, this->actor.floorPoly, this->actor.floorBgId) == 1)) {
+                this->actor.speedXZ = CLAMP_MAX(temp_f2, 16.0f);
+                this->actor.world.rot.y = Math_FAtan2F(temp_f12, temp_f14);
+            }
+
+            if (!Math_StepToF(&this->actor.speedXZ, 0.0f, 1.0f)) {
+                direction = this->actor.world.rot.y;
+                if (ABS_ALT((s16)(this->actor.world.rot.y - this->actor.shape.rot.y)) > 0x4000) {
```

Comment:
```suggestion
                if (ABS_ALT(BINANG_SUB(this->actor.world.rot.y, this->actor.shape.rot.y)) > 0x4000) {
```
Assuming this works, looks cleaner imo, but up to you if you want to use it.